libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx

```diff
@@ -0,0 +1,101 @@
+import { type GitProviderEnum } from 'qovery-typescript-axios'
+import { Controller, useFormContext } from 'react-hook-form'
+import { useParams } from 'react-router-dom'
+import { InputSelect, InputText, LoaderSpinner } from '@qovery/shared/ui'
+import { getGitTokenValue } from '../git-provider-setting/git-provider-setting'
+import useBranches from '../hooks/use-branches/use-branches'
+
+export interface GitBranchSettingProps {
+  gitProvider: GitProviderEnum | string
+  disabled?: boolean
+}
+
+export function GitBranchSetting({ disabled, gitProvider }: GitBranchSettingProps) {
+  const { control, watch } = useFormContext()
+  const { organizationId = '' } = useParams()
+
+  const getGitToken = getGitTokenValue(gitProvider)
+  const provider = getGitToken ? getGitToken?.type : gitProvider
+  const watchFieldRepository = watch('repository')
+  const watchFieldBranch = watch('branch')
+
+  const {
+    data: branches = [],
+    isError,
+    isLoading,
+    isRefetching,
+  } = useBranches({
+    organizationId,
+    gitProvider: provider,
+    name: watchFieldRepository,
+    gitToken: getGitToken?.id,
+    enabled: !disabled,
+  })
+
+  if (isError) {
+    return null
+  }
+
+  if (!disabled && (isLoading || isRefetching)) {
+    return (
+      <div data-testid="loader" className="flex justify-center">
+        <LoaderSpinner />
+      </div>
+    )
+  } else {
```

Comment:
no need for `else` with `if` with early returns

Reply:
https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55